src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -584,23 +579,38 @@ public static CSharpCodeWritingScope BuildClassDeclaration(
         }
 
         return new CSharpCodeWritingScope(writer);
-    }
 
-    public static void WriteWithPragma(this CodeWriter writer, string content, CodeRenderingContext context, SourceSpan source)
-    {
-        if (context.Options.DesignTime)
+        void writeOptionalToken(IntermediateToken token)
         {
-            using (writer.BuildLinePragma(source, context))
+            if (token is not null)
             {
-                context.AddSourceMappingFor(source);
-                writer.Write(content);
+                if (token.Source is { } source)
+                {
+                    WriteWithPragma(writer, token.Content, context, source);
+                }
+                else
+                {
+                    writer.Write(token.Content);
+                }
             }
         }
-        else
+
+        static void WriteWithPragma(CodeWriter writer, string content, CodeRenderingContext context, SourceSpan source)
```

Comment:
Inconsistent naming: local function `writeOptionalToken` has the first letter lower case, this one has the first letter uppercase.